eng/scripts/Compare-CurrentToCodegeneration.ps1

```diff
@@ -17,55 +17,81 @@ Azure SDK for Java repository. CI jobs should use the 'ServiceDirectory', such a
 
 param(
   [Parameter(Mandatory = $false)]
-  [string]$Directory
+  [string]$ServiceDirectories
 )
 
-$path = ""
-if ($Directory) {
-  $path = $Directory
-}
+# Returns true if there's an error, false otherwise
+function Compare-CurrentToCodegeneration {
+  param(
+    [Parameter(Mandatory=$true)]
+    $ServiceDirectory
+  )
 
-$swaggers = Get-ChildItem -Path $path -Filter "Update-Codegeneration.ps1" -Recurse
-if ($swaggers.Count -eq 0) {
-  Write-Host "
+  $swaggers = Get-ChildItem -Path $ServiceDirectory -Filter "Update-Codegeneration.ps1" -Recurse
+  if ($swaggers.Count -eq 0) {
+    Write-Host "
 
-  ===========================================
-  No Swagger files to regenerate
-  ===========================================
+    =========================================================
+    No Swagger files to regenerate for $ServiceDirectory
+    =========================================================
 
-  "
-  exit 0
-}
+    "
+    return $false
+  }
 
 
-Write-Host "
+  Write-Host "
 
-===================================
-Invoking Autorest code regeneration
-===================================
+  =========================================================
+  Invoking Autorest code regeneration for $ServiceDirectory
+  =========================================================
 
-"
+  "
 
-foreach ($script in $swaggers) {
-  Invoke-Expression $script.FullName
-}
+  foreach ($script in $swaggers) {
+    Write-Host "Calling Invoke-Expression $($script.FullName)"
+    Invoke-Expression $script.FullName
```

Comment:
This being in a function appears to break console logging now.

This change makes it work fine:

`(& $script.FullName) | Write-Host`